cfme/ansible_tower/jobs.py

```diff
@@ -38,15 +42,101 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(View):
+    title = Text('//div[@id="main-content"]//h1')
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download')
+    search = View.nested(Search)
+
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        title = '{} (Summary)'.format(self.obj.name)
+        return self.title.text == title
+
+
 @attr.s
 class TowerJobs(BaseEntity):
-    pass
+    # pass
+
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self.parent, 'All')
+        job_status = ''
+        for row in view.entities.elements:
+            print(row['Template Name'].read())
+            if row['Template Name'].read() == self.template_name and job_status == '':
+                job_status = row.status.text
+            else:
+                pass
+        return job_status
+
+    def delete(self):
+        view = navigate_to(self.parent, 'All')
+        view.configuration.item_select('Remove Jobs', handle_alert=True)
 
 
 @attr.s
 class TowerJobsCollection(BaseCollection):
     ENTITY = TowerJobs
 
+    def all(self):
+        # a.collections.ansible_tower_jobs.filter({"id":'1469'})
+        id = self.filters.get('id', 0)
+        teml_name = self.filters.get('temp_name')
+        view = navigate_to(self, 'All')
+        jobs = []
+        # for row in view.items.rows():
+        for _ in view.entities.paginator.pages():
+            for row in view.entities.elements:
+                if int(row.id.text) > int(id):
+                    if teml_name == row.template_name.text:
+                        jobs.append(self.instantiate(template_name=row.template_name.text))
+        return jobs
+
+    def first_by_date(self):
+        view = navigate_to(self, 'All')
+        # view.items.sort_by(column='Created On', order='asc')
+        view.entities.elements.sort_by(column='Created On', order='asc')
+        # row = view.items[0]
+        row = view.entities.elements[0]
+        job = self.instantiate(template_name=row.template_name.text)
+        return job
+
+    def is_all_finished(self):
+        jobs = self.all()
+        for job in jobs:
+            if job.status == "successfull":
+                return True
```

Comment:
Required:
This will match first that success and stop checking other jobs
```
all_success = True
for job in jobs:
    if job.status == "successfull":
         continue
    else:
         all_success = False
return all_success